src/fread.c

```diff
@@ -910,7 +911,7 @@ static void parse_double_hexadecimal(FieldParseContext *ctx)
     acc <<= (13 - ndigits) * 4;
     ch += 1 + (Eneg = ch[1]=='-') + (ch[1]=='+');
     uint64_t E = 0;
-    while ((digit = (uint8_t)(*ch-'0')) < 10) {
+    while ((digit = AS_DIGIT(*ch)) < 10) {
```

Comment:
note the difference of `uint8_t` --> `uint_fast8_t`. I assume `uint8_t` was unintentional